src/wallet/monero_wallet_light.cpp

```diff
@@ -0,0 +1,3706 @@
+    cryptonote::blobdata blob;
+    if (!epee::string_tools::parse_hexstr_to_binbuff(hex, blob))
+    {
+      throw std::runtime_error("Failed to parse hex.");
+    }
+
+    bool loaded = false;
+    tools::wallet2::pending_tx ptx;
+
+    try
+    {
+      binary_archive<false> ar{epee::strspan<std::uint8_t>(blob)};
+      if (::serialization::serialize(ar, ptx))
+        loaded = true;
+    }
+    catch(...) {}
+
+    if (!loaded)
+    {
+      try
+      {
+        std::istringstream iss(blob);
+        boost::archive::portable_binary_iarchive ar(iss);
+        ar >> ptx;
+      }
+      catch (...) {
+        throw std::runtime_error("Failed to parse tx metadata.");
+      }
+    }
+
+    return epee::string_tools::pod_to_hex(cryptonote::get_transaction_hash(ptx.tx));
+  }
+
+  // ------------------------------- DESERIALIZE UTILS -------------------------------
+
+  std::shared_ptr<monero_light_output> monero_light_output::deserialize(const std::string& config_json) {
+    // deserialize monero output json to property node
+    std::istringstream iss = config_json.empty() ? std::istringstream() : std::istringstream(config_json);
+    boost::property_tree::ptree node;
+    boost::property_tree::read_json(iss, node);
+
+    // convert config property tree to monero_wallet_config
+    std::shared_ptr<monero_light_output> output = std::make_shared<monero_light_output>();
+    
+    for (boost::property_tree::ptree::const_iterator it = node.begin(); it != node.end(); ++it) {
+        std::string key = it->first;
+
+        if (key == std::string("tx_id")) output->m_tx_id = it->second.get_value<uint64_t>();
+        else if (key == std::string("amount")) output->m_amount = it->second.data();
+        else if (key == std::string("index")) output->m_index = it->second.get_value<uint16_t>();
+        else if (key == std::string("global_index")) output->m_global_index = it->second.data();
+        else if (key == std::string("rct")) output->m_rct = it->second.data();
+        else if (key == std::string("tx_hash")) output->m_tx_hash = it->second.data();
+        else if (key == std::string("tx_prefix_hash")) output->m_tx_prefix_hash = it->second.data();
+        else if (key == std::string("public_key")) output->m_public_key = it->second.data();
+        else if (key == std::string("tx_pub_key")) output->m_tx_pub_key = it->second.data();
+        else if (key == std::string("spend_key_images")) for (boost::property_tree::ptree::const_iterator it2 = it->second.begin(); it2 != it->second.end(); ++it2) output->m_spend_key_images.get().push_back(it2->second.data());
+        else if (key == std::string("timestamp")) output->m_timestamp = it->second.data();
+        else if (key == std::string("height")) output->m_height = it->second.get_value<uint64_t>();
+    }
+
+    return output;
+  }
+
+  std::shared_ptr<monero_light_rates> monero_light_rates::deserialize(const std::string& config_json) {
```

Comment:
It could be useful to calculate the balance in local currency, maybe exposing a method like `float monero_wallet_light::get_rate(string fiat_code)`

Reply:
> It could be useful to calculate the balance in local currency, maybe exposing a method like `float monero_wallet_light::get_rate(string fiat_code)`

Where are the rates coming from, the light wallet server? I ask because typically pricing can be a service on top of a pure native XMR wallet, and not part of the wallet api itself, which creates a dependence on external services.

Reply:
The rates object comes from lws server, in particularly in the get_address_info method